lib/ansible/plugins/callback/slack_github.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import os
+import subprocess
+from ansible.module_utils.urls import open_url
+from ansible.plugins.callback import CallbackBase
+
+
+class CallbackModule(CallbackBase):
+    """This is an ansible callback plugin that sends full
+    play recap to github gist and sends colored notification
+    to  a Slack channel after play ends.
+
+    This plugin makes use of file `slack.json` in your playbook folder
+    with following contents:
+    ```json
+    {
+        "webhook": "Slack incoming webhook url",
+        "username": "Ansible"
+    }
+    ```
+
+    Output: https://gist.github.com/anonymous/f75e04a115ea86b730e41e2ef03cc5f0
+    """
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'notification'
+    CALLBACK_NAME = 'slack_github'
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self, display=None):
+        self.disabled = False
+        super(CallbackModule, self).__init__(display=display)
+        self.github_api = 'https://api.github.com/'
+        self._log = []
+
+    def init_config(self, playbook):
+        self.playbook_name = os.path.basename(playbook._file_name)
+        config_file = os.path.join(
+            playbook._basedir.encode(
+                'ascii', 'ignore'), 'slack.json')
+        if not os.path.isfile(config_file):
+            self._display.warning(
+                'File {} not found. SlackGithub callback has been disabled'.format(config_file))
+            self.disabled = True
+            return
+        with open(config_file) as config_file:
+            self.config = json.load(config_file)
+        try:
+            cmd = ['git', 'config', '--get', 'user.name']
+            username = subprocess.check_output(
+                cmd, universal_newlines=True).split('\0')
+            self.user = username[0].rstrip()
+        except BaseException:
+            self.user = 'Anonymous'
+
+    def log(self, text):
+        self._log.append(text)
+
+    def send(self, color='good'):
+        if self.disabled:
+            return
+        raw_gist = {
+            'description': 'Ansible play recap',
+            'public': False,
+            'files': {
+                'recap.md': {
+                    'content': '\n'.join(
+                        self._log)}}}
+
+        data = json.dumps(raw_gist)
+        try:
+            response = open_url(
+                self.github_api + 'gists',
+                data=data,
+                method='POST')
+        except Exception as e:
+            self._display.warning('Could not submit play recap: {}'.format(e))
+
+        gist = json.loads(response.read())
+        url = gist[u'html_url'].encode('ascii', 'ignore')
+        hosts = '*{}*'.format(', '.join(self.play.hosts))
```

Comment:
Note that format strings changed in python-2.7  So things like "{} {}".format() needs to change to "{0} {1}".format() for python-2.6 compatibility.